src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
+     *
+     * @param classSet {@link Set<Class>}
+     */
+    private void registerBeansInContext(Set<Class<?>> classSet) throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        for (Class<?> beanClass : classSet) {
+            var constructor = beanClass.getConstructor();
+            var beanInstance = constructor.newInstance();
+            String beanName = resolveBeanName(beanClass);
             try {
-                context.put(beanName, bean.getDeclaredConstructor().newInstance());
-            } catch (InstantiationException | InvocationTargetException | IllegalAccessException |
-                     NoSuchMethodException e) {
+                context.put(beanName, beanInstance);
+            } catch (Exception e) {
                 LOGGER.error("Can't create an instance of {} class", beanName);
                 throw new RuntimeException(e);
             }
+
         }
     }
 
     /**
-     * Store bean in the context
+     * <h3>Injects Beans in Autowired field</h3>
+     */
+    private void autoWireBeans() throws IllegalAccessException, NoSuchFieldException {
+        for (Map.Entry<String, Object> entry : context.entrySet()) {
+            var beanInstance = entry.getValue();
+            injectBeanViaAutowiredAnnotation(beanInstance.getClass(), beanInstance);
+            injectBeanViaConstructor(beanInstance.getClass(), beanInstance);
+        }
+    }
+
+    /**
+     * <h3>Injects Beans via {@link Autowired} annotation</h3>
+     *
+     * @param beanClass    {@link Class}
+     * @param beanInstance {@link Object}
+     */
+    private <T> void injectBeanViaAutowiredAnnotation(Class<?> beanClass, T beanInstance) throws IllegalAccessException {
+        for (Field field : beanClass.getDeclaredFields()) {
+            field.setAccessible(true);
+            if (field.isAnnotationPresent(Autowired.class)) {
+                var autowiredBeansInstance = getBean(field.getType());
+                field.set(beanInstance, autowiredBeansInstance);
+            }
+        }
+    }
+    /**
+     * <h3>Injects Bean via {@link Constructor}</h3>
+     *
+     * @param beanClass    {@link Class}
+     * @param beanInstance {@link Object}
+     */
+    private <T> void injectBeanViaConstructor(Class<?> beanClass, T beanInstance) throws NoSuchFieldException, IllegalAccessException {
+        Map<Class<?>, String> fieldTypeStringNameMap = new HashMap<>();
+        for (Field field : PrinterServiceDemo.class.getDeclaredFields()){
+            fieldTypeStringNameMap.put(field.getType(), field.getName());
+        }
+        Constructor<?>[] constructors = beanClass.getConstructors();
+        for (Constructor<?> constructor : constructors){
+            for (Parameter parameter: constructor.getParameters()){
+                Class<?> parameterType  = parameter.getType();
+                var autowiredBeansInstance = getBean(parameterType);
```

Comment:
Very complicated name `autowiredBeansInstance` we call getBean method and want to get bean. I think `var bean`  will be sufficient